datafusion/physical-expr/src/window/built_in.rs

```diff
@@ -245,8 +241,40 @@ impl WindowExpr for BuiltInWindowExpr {
         // NOTE: Currently, groups queries do not support the bounded memory variant.
         self.expr.supports_bounded_execution()
             && (!self.expr.uses_window_frame()
-                || !(self.window_frame.start_bound.is_unbounded()
-                    || self.window_frame.end_bound.is_unbounded()
+                || !(self.window_frame.end_bound.is_unbounded()
                     || matches!(self.window_frame.units, WindowFrameUnits::Groups)))
     }
 }
+
+// When the window frame has a fixed beginning (e.g UNBOUNDED PRECEDING), we
+// can prune our datasets for FIRST_VALUE, LAST_VALUE and NTH_VALUE functions.
+// Once the result is final, it will be always stay the same. Hence, we do not
+// need to keep past data as we process the entire dataset.
+fn prune_nth_value(
```

Comment:
Minor suggestion is to call this "memoize_nth_value" or something rather than `prune` as I think if pruning in the context of pruning predicates / removing rows